Polyhedron/demo/Polyhedron/Plugins/Point_set/Point_set_upsampling_plugin.cpp

```diff
@@ -125,10 +129,15 @@ void Polyhedron_demo_point_set_upsampling_plugin::on_actionEdgeAwareUpsampling_t
 					  dialog.edge_sensitivity(),
 					  dialog.neighborhood_radius() * average_spacing,
 					  output_size);
-
+      nb_selected += new_points.size();
+      
       for (unsigned int i = 0; i < new_points.size (); ++ i)
-	points->push_back (Point_set::Point_with_normal (new_points[i].first,
-							 new_points[i].second));
+        {
+          points->push_back (Point_set::Point_with_normal (new_points[i].first,
+                                                           new_points[i].second));
+        }
+      if (nb_selected != new_points.size())
+        points->set_first_selected (points->end() - nb_selected);
```

Comment:
I find this strange. From what I understand, you add new points to the selection so first does not change.